pkg/virt-api/webhooks/arm64.go

```diff
@@ -0,0 +1,157 @@
+				}, false
+			}
+		}
+	}
+	if &spec.Domain.Devices != nil && spec.Domain.Devices.AutoattachGraphicsDevice != nil && *spec.Domain.Devices.AutoattachGraphicsDevice {
+		return metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueNotSupported,
+			Message: "VGA device is not support by buildin qemu-kvm in virt-launcher image for Arm64, please disable autoattachGraphicsDevice",
+			Field:   field.Child("domain", "device", "autoattachgraphicsdevice").String(),
+		}, false
+	}
+	if spec.Domain.CPU != nil && (&spec.Domain.CPU.Model != nil) && spec.Domain.CPU.Model == "host-model" {
+		return metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueNotSupported,
+			Message: "Arm64 not support host model well",
+			Field:   field.Child("domain", "cpu", "model").String(),
+		}, false
+	}
+	return metav1.StatusCause{}, true
+}
+
+// defaultVerifyArchFunc verify if arch is arm64
+func defaultVerifyArchFunc() bool {
+	if rt.GOARCH == "arm64" {
+		return true
+	}
+	return false
+}
+
+// setDefaultCPUModel set default cpu model to host-passthrough
+func setDefaultCPUModel(vmi *v1.VirtualMachineInstance) {
+	//if vmi doesn't have cpu topology or cpu model set
+	if vmi.Spec.Domain.CPU == nil || vmi.Spec.Domain.CPU.Model == "" {
+		// create cpu topology struct
+		if vmi.Spec.Domain.CPU == nil {
+			vmi.Spec.Domain.CPU = &v1.CPU{}
+		}
+		//set is as vmi cpu model
+		if vmi.Spec.Domain.CPU.Model == "" {
+			vmi.Spec.Domain.CPU.Model = defaultCPUModel
+		}
+	}
+}
+
+// setDefaultDevices set AutoattachGraphicsDevice to false
+func setDefaultDevices(vmi *v1.VirtualMachineInstance) {
+	if &vmi.Spec.Domain.Devices == nil || vmi.Spec.Domain.Devices.AutoattachGraphicsDevice == nil {
+		if &vmi.Spec.Domain.Devices == nil {
+			vmi.Spec.Domain.Devices = v1.Devices{}
+		}
+		vmi.Spec.Domain.Devices.AutoattachGraphicsDevice = &_false
+	}
+}
+
+// setDefaultBootloader set default bootloader to uefi boot
+func setDefaultBootloader(vmi *v1.VirtualMachineInstance) {
+	if vmi.Spec.Domain.Firmware == nil || vmi.Spec.Domain.Firmware.Bootloader == nil {
+		if vmi.Spec.Domain.Firmware == nil {
+			vmi.Spec.Domain.Firmware = &v1.Firmware{}
+		}
+		if vmi.Spec.Domain.Firmware.Bootloader == nil {
+			vmi.Spec.Domain.Firmware.Bootloader = &v1.Bootloader{}
+		}
+		vmi.Spec.Domain.Firmware.Bootloader.EFI = &v1.EFI{}
+		vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot = &_false
+	}
+}
+
+// ValidateVirtualMachineInstanceArm64Setting is validation function for validating-webhook
+func ValidateVirtualMachineInstanceArm64Setting(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+
+	if VerifyArchFunc() {
+		if cause, ok := verifyInvalidSetting(field, spec); !ok {
+			causes = append(causes, cause)
+		}
+	}
+
+	return causes
+}
+
+// SetVirtualMachineInstanceArm64Setting is mutating function for mutating-webhook
+func SetVirtualMachineInstanceArm64Setting(vmi *v1.VirtualMachineInstance) error {
```

Comment:
would it make sense to rename ``SetVirtualMachineInstanceArm64Setting``` as ```SetVirtualMachineInstanceArm64Defaults``` ? that might better align with the other default functions in the mutation webhook